packages/next/server/request-meta.ts

```diff
@@ -53,16 +53,43 @@ export function addRequestMeta<K extends keyof RequestMeta>(
   return setRequestMeta(request, meta)
 }
 
-export type NextParsedUrlQuery = ParsedUrlQuery & {
+type NextQueryMetadata = {
   __nextDefaultLocale?: string
   __nextFallback?: 'true'
   __nextLocale?: string
   __nextSsgPath?: string
   _nextBubbleNoFallback?: '1'
   _nextDataReq?: '1'
-  amp?: '1'
 }
 
+export type NextParsedUrlQuery = ParsedUrlQuery &
+  NextQueryMetadata & {
+    amp?: '1'
```

Comment:
I'm not considering `amp` as a Next.js metadata, so we won't merge that automatically in middlewares. I am not sure if this is the right thing to do, but it was a hunch